src/svc_auth_gss.c

```diff
@@ -385,6 +385,63 @@ svcauth_gss_nextverf(struct svc_req *req, struct svc_rpc_gss_data *gd,
 	return (true);
 }
 
+static bool
+gss_check_seq_num_valid(struct svc_rpc_gss_data *gd, int seq_num)
+{
+	int bit_to_clear;
+	__warnx(TIRPC_DEBUG_FLAG_RPCSEC_GSS,
+		"seq %d seqlast %d diff %d bit# %d",
+		seq_num, gd->seqlast, seq_num - gd->seqlast,
+		seq_num % GSS_SEQ_WIN);
+
+	/* According to rfc2203, sequence numbers should be less than:
+	 * MAXSEQ 0x80000000.
+	 */
+	if (seq_num < 0)
+		return false;
```

Comment:
This case is actually supposed to return an error to the client:
   When the client's sequence number exceeds the maximum the server will
   allow, the server will reject the request with the reason
   RPCSEC_GSS_CTXPROBLEM.

I suspect this means that a bool is not sufficient for the return value of this function.  Alternatively, this check can be done before this function is called.

Reply:
Makes sense.